packages/hierarchy-builder/src/hierarchy-builder/queries/NodeSelectQueryFactory.ts

```diff
@@ -86,20 +80,29 @@ export interface NodeSelectClauseProps {
   hideNodeInHierarchy?: boolean | ECSqlValueSelector;
   hideIfNoChildren?: boolean | ECSqlValueSelector;
   grouping?: ECSqlSelectClauseGroupingParams;
-  mergeByLabelId?: string | ECSqlValueSelector;
 }
 
 /**
  * A data structure for defining nodes' grouping requirements.
  * @beta
  */
 export interface ECSqlSelectClauseGroupingParams {
-  byLabel?: boolean | ECSqlSelectClauseGroupingParamsBase | ECSqlValueSelector;
+  byLabel?: ECSqlSelectClauseLabelGroupingParams;
   byClass?: boolean | ECSqlSelectClauseGroupingParamsBase | ECSqlValueSelector;
   byBaseClasses?: ECSqlSelectClauseBaseClassGroupingParams;
   byProperties?: ECSqlSelectClausePropertiesGroupingParams;
 }
 
+/**
+ * A data structure for defining label grouping.
+ * @beta
+ */
+export type ECSqlSelectClauseLabelGroupingParams =
+  | { mergeId: string | ECSqlValueSelector }
+  | boolean
+  | ECSqlSelectClauseGroupingParamsBase
+  | ECSqlValueSelector;
```

Comment:
What do you think about changing this to:
```ts
interface ECSqlSelectClauseLabelGroupingBaseParams {
  // Defaults to "group"
  action?: "group" | "merge";
  // Specifying different group IDs allows us to avoid grouping different kinds of nodes. For example, we may not want to group Model and Element nodes together, even if their labels are the same.
  groupId?: "string";
}
interface ECSqlSelectClauseLabelGroupingMergeParams extends ECSqlSelectClauseLabelGroupingBaseParams {
  action: "merge";
}
interface: ECSqlSelectClauseLabelGroupingGroupParams extends ECSqlSelectClauseLabelGroupingBaseParams, ECSqlSelectClauseGroupingParamsBase {
  action?: "group";
}
type ECSqlSelectClauseLabelGroupingParams =
  | boolean
  | ECSqlValueSelector
  | ECSqlSelectClauseLabelGroupingMergeParams 
  | ECSqlSelectClauseLabelGroupingGroupParams;
```

I believe this kind of API will be easier to document and understand, since we'll have an `action` property which tells how we're going to group the nodes. 

Also, it makes the `mergeId` / `groupId` optional and shares it for both - merging and grouping cases. 
- We need this attribute only when there are multiple different types of nodes at the same hierarchy level and we don't want them to be merged together, which is kind of a rare case. Making it optional makes the general case simpler. 
- We have cases where we don't want to merge nodes together due to their differences. Seems like the same feature would be beneficial for grouping as well.

Reply:
Your suggestion eliminates the `byLabel: true` and `byLabel: { selector: "x.SomeProperty" }` specification types. But, based on the fact that label grouping basically requires also specifying additional attributes (`hideIfNoSiblings` & `hideIfOneGroupedNode`), I think it's fine to eliminate them.

Regarding the `action: "group"` attribute, I think it should stay optional for when we want grouping. The context (you access these props through `node.processingParams.grouping.byLabel`) already suggests we're talking about grouping, so for the most common scenario I don't think we need to repeat that.